lib/juttle-service-client.js

```diff
@@ -0,0 +1,290 @@
+'use strict';
+
+/* eslint no-console: 0 */
+var _ = require('underscore');
+var Promise = require('bluebird');
+var WebSocket = require('ws');
+var minimist = require('minimist');
+var rp = require('request-promise');
+var path = require('path');
+var cli_errors = require('juttle/lib/cli/errors');
+var rp_errors = require('request-promise/errors');
+var JuttleBundler = require('../lib/bundler');
+var fs = require('fs');
+var crypto = require('crypto');
+
+
+// service_option is the name of the argument for the service
+// (juttle-engine or juttle-service). addl_commands/addl_options can
+// be provided to include additional commands/options on top of the
+// commands/options supported here.
+
+let service_option = 'juttle-service';
+let addl_commands = [];
+let addl_options = [];
+
+function init(options) {
+    service_option = options.service_option;
+    addl_commands = options.addl_commands;
+    addl_options = options.addl_options;
+}
+
+function usage() {
```

Comment:
The usage is tightly coupled with the argument parsing so it seems to me it belongs in the script itself. I'd suspect that the future `juttle-engine-client` will have different (albeit highly similar) usage. Still that repetition seems like a necessary evil since, as you said, the engine variant will include support for rendezvous.